src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,269 @@
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Start event)
+    {
+        Log.getLogger().info("ExplosionEvent.Start");
+        if (colony.isCoordInColony(event.getWorld(), new BlockPos(event.getExplosion().getPosition())))
+        {
+            Log.getLogger().info("Explosion in colony, removing!");
+            cancelEvent(event);
+        }
+
+    }
+
+    /**
+     * PlayerInteractEvent handler.
+     * <p>
+     * Check, if a player right clicked a block.
+     * Deny if:
+     * - If the block is in colony
+     * - block is AbstractBlockHut
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        Log.getLogger().info("Check if coordinate is in colony `" + this.colony.getName() + "`");
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            Log.getLogger().info("Rightclick Coordinate is in `" + this.colony.getName() + "`");
+
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+
+            // Huts
+            if (block instanceof AbstractBlockHut &&
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
+     * TODO Check this behavior
+     */
+    @SubscribeEvent
+    public void on(final PlayerContainerEvent.Open event)
+    {
+        if (this.colony.isCoordInColony(event.getEntity().getEntityWorld(), event.getEntity().getPosition()))
+        {
+            if (this.colony.getPermissions().isColonyMember(event.getEntityPlayer()))
+            {
+                Log.getLogger().info("Colony member `" + event.getEntityPlayer().getName() + "` opens container `" + event.getEntity().getName() + "`");
+            }
+            else
+            {
+                Log.getLogger().info("Player `" + event.getEntity().getName() + "` is not a member of `" + this.colony.getName() + "`");
+            }
+        }
+        else
+        {
+            Log.getLogger().info("The opened entity is not inside `" + this.colony.getName() + "`");
+        }
+    }
+
+    /**
+     * ItemTossEvent handler.
+     *
+     * @param event ItemTossEvent
+     */
+    @SubscribeEvent
+    public void on(ItemTossEvent event)
+    {
+        final EntityPlayer playerIn = event.getPlayer();
+
+        Log.getLogger().info(String.format("Check if playerIn `%s` is inside colony `%s`", playerIn.getName(), colony.getName()));
+        if (colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            Log.getLogger().info(String.format("Check if playerIn `%s` has at least Rank `%s`", playerIn.getName(), colony.getName()));
+
+            Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'rank' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)